### DISCUSSION TITLE
Allow Editing of Helpers in Both GUI and YAML

### DISCUSSION THREAD

**Discussion by MAFL74** [home-assistant/architecture#1170] [General] [answered]
Feature Request: Enable editing of Home Assistant helpers (input_boolean, input_text, input_number, etc.) in both the GUI and YAML, similar to how automations can be edited.

Problem Statement: Currently, helpers created via the GUI are stored in a hidden database and cannot be directly edited in YAML. This creates an inconsistency, as automations can be freely edited in both GUI and YAML. For advanced users who prefer version control, YAML editing, and the flexibility it provides, this limitation is a significant inconvenience.

Proposed Solution:

Allow users to edit helpers in YAML after they are created in the GUI.
Similar to automations, provide a "Edit in YAML" option for helpers.
Store helpers in a separate YAML file if edited manually, while still allowing GUI-based management.
Benefits:

Consistency: Aligns with how automations are currently handled, providing a consistent experience.
Flexibility: Advanced users can leverage version control (e.g., Git) and bulk changes.
Efficiency: YAML is faster for making large or complex changes.
Debugging: Easier to troubleshoot and review configurations.

**Comment by parautenbach** [comment, marked as answer]
Please see [What is the architecture repository for?](https://github.com/home-assistant/architecture/discussions/494) and head over to the [FR category](https://community.home-assistant.io/c/feature-requests/13) on the forum. 

> If you raise an architectural proposal, your intention should be to implement the proposal when accepted. In case you are not able to implement your proposal (for whatever reason), a feature request on our forum should be raised instead.
